transform/runner.go

```diff
@@ -38,21 +48,83 @@ func (r *Runner) Run(object unstructured.Unstructured, plugins []Plugin) ([]byte
 			patches = append(patches, resp.Patches...)
 		}
 	}
+	response := RunnerResponse{}
+
 	// TODO: in the future we should consider a way to speed this up with go routines.
 	if len(errs) > 0 {
 		// TODO: handle error in a reasonable way. Probably needs an enhancement
 		// Should Consider option to ignore errors
-		return nil, false, errs[0]
+		return response, errs[0]
 	}
 	if haveWhiteOut {
 		// TODO: handle if we should skip whiteOut if there is a transform
-		return nil, true, nil
+		response.HaveWhiteOut = haveWhiteOut
+		return response, nil
 	}
+
 	if havePatches {
-		// TODO: Handle dedup
-		// TODO: Handle conflicts with paths
-		b, err := json.Marshal(patches)
-		return b, false, err
+		patches, ignoredPatches, err := r.removeDuplicates(patches)
+		if err != nil {
+			return response, err
+		}
+
+		// for each patch, we should make sure the patch can be applied
+		// We may need to break the transform file into two parts to handle this correctly
+		if len(patches) != 0 {
+			response.TransformFile, err = json.Marshal(patches)
+			if err != nil {
+				return response, err
+			}
+		}
+		if len(ignoredPatches) != 0 {
+			response.IgnoredPatches, err = json.Marshal(ignoredPatches)
+			if err != nil {
+				return response, err
+			}
+		}
+
+		return response, err
+	}
+	return response, nil
+}
+
+type operatationKey struct {
+	Kind string
+	Path string
+}
+
+func (r *Runner) removeDuplicates(patch jsonpatch.Patch) (jsonpatch.Patch, jsonpatch.Patch, error) {
```

Comment:
I would rename this to make it clear that we're not only removing duplicates but also removing/ignoring clashing patches.